### PR TITLE
iOS E2E Testing + Improving E2E Code-Reuse

### PR DESCRIPTION
**Pull request by samuelfreiberg**
### Platforms Impacted
- [X] iOS
- [X] macOS
- [X] win32 (Office)
- [X] windows
- [ ] android

### Description of changes

This PR does a couple things:

1. Adds iOS E2E Testing and integrates within out CI
2. Improves code-reuse + cross-platformability of our E2E Testing infrastructure. Now, we only need **one page object per component**. This could change in the future when the scope of testing is built out further. We can do this because we can use the same selector cross-platform (I previously didn't think we could), thus, we don't need separate page objects to import separate selectors.

### Verification

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| Screenshot or description before this change | Screenshot or description with this change |

### Pull request checklist

This PR has considered (when applicable):
- [ ] Automated Tests
- [ ] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

File: apps/win32/wdio.conf.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ const appArgs = 'basePath ' + path.resolve('dist') + ' plugin defaultplugin bund
 const appDir = path.dirname(require.resolve('@office-iss/rex-win32/rex-win32.js'));
 
 const defaultWaitForTimeout = 20000;
-const defaultConnectionRetryTimeout = 20000;
+const defaultConnectionRetryTimeout = 60000;
```
Comment (ksiler) [inline apps/win32/wdio.conf.js:9]: Is there a reason for the time increase?

**Comment by ksiler** [comment]
I noticed the pipeline has a couple of tasks called "Copy non-Win32 Appium Reports/tree dump screenshots" when copying reports/screenshots for platforms that aren't win32 (in [e2e-publish-artifacts.yml](https://github.com/microsoft/fluentui-react-native/blob/main/.ado/templates/e2e-publish-artifacts.yml#L14)). Could we make this "Copy ${{ parameters.applicationType }} Appium Reports/tree dump screenshots" instead?

Awesome work by the way! Glad to see the added improvements.

**Comment by samuelfreiberg** [comment]
> I noticed the pipeline has a couple of tasks called "Copy non-Win32 Appium Reports/tree dump screenshots" when copying reports/screenshots for platforms that aren't win32 (in [e2e-publish-artifacts.yml](https://github.com/microsoft/fluentui-react-native/blob/main/.ado/templates/e2e-publish-artifacts.yml#L14)). Could we make this "Copy ${{ parameters.applicationType }} Appium Reports/tree dump screenshots" instead?
> 
> Awesome work by the way! Glad to see the added improvements.

Yes, definitely! Just pushed the changes, thanks for the improvement!

File: apps/win32/wdio.conf.js (hunk 1 above)
Comment (samuelfreiberg) [reply apps/win32/wdio.conf.js:9]: I've noticed the first test had to re-run a couple times over the past week due to a failure, so I think increasing this timer will give WDIO more time to connect to the app

**Review by ksiler — APPROVED** [review]

**Comment by Saadnajmi** [comment]
:o iOS E2E tests work now!

I guess the ".apple" naming scheme didn't work?

**Comment by samuelfreiberg** [comment]
> :o iOS E2E tests work now!
> 
> I guess the ".apple" naming scheme didn't work?

Yes, it's working!

The reason I couldn't do the .apple convention like I mentioned is because iOS has a different way of scrolling than macOS (because it's mobile). So they couldn't have the same .spec file (which is why I used .ios and .macos again).

**Review by Saadnajmi — APPROVED** [review]

**Review by chiuam — APPROVED** [review]

**Comment by Saadnajmi** [comment]
Would it be possible to not need these two "copy win32" checks?

image

**Comment by samuelfreiberg** [comment]
> Would it be possible to not need these two "copy win32" checks?
> 
> image

Fixed :)

**Pull request merged**